support-models/src/main/scala/com/gu/support/workers/PaymentFields.scala

```diff
@@ -11,19 +11,34 @@ sealed trait PaymentFields
 
 case class PayPalPaymentFields(baid: String) extends PaymentFields
 
-sealed trait StripePaymentFields extends PaymentFields {
-  val stripePaymentType: Option[StripePaymentType]
-}
-
-case class StripeSourcePaymentFields(
-    stripeToken: String,
-    stripePaymentType: Option[StripePaymentType],
-) extends StripePaymentFields // pre SCA compatibility
-
-case class StripePaymentMethodPaymentFields(
+case class StripePaymentFields(
     paymentMethod: PaymentMethodId,
     stripePaymentType: Option[StripePaymentType],
-) extends StripePaymentFields
+    stripePublicKey: Option[StripePublicKey], // this is only optional until all checkouts are updated
+) extends PaymentFields
+
+object StripePublicKey {
+
+  def parse(value: String): Either[String, StripePublicKey] =
+    if (
+      value.length > 10 &&
+      value.forall { char =>
+        (char >= 'a' && char <= 'z') ||
+        (char >= 'A' && char <= 'Z') ||
+        (char >= '0' && char <= '9') ||
+        char == '_'
+      }
+    ) Right(new StripePublicKey(value))
+    else Left(s"StripePublicKey <$value> had invalid characters")
+
+  def get(value: String): StripePublicKey =
+    parse(value).left.map(new Throwable(_)).toTry.get
```

Comment:
by converting to a try before doing get, you get the actual error message in the exception, rather than just a generic None.get exception

Reply:
Could this comment exist next to the code for the next engineer that passes through?

Reply:
good idea, done that now